patches/minecraft/net/minecraft/client/renderer/ItemBlockRenderTypes.java.patch

```diff
@@ -42,74 +42,91 @@
           if (!Minecraft.m_91085_()) {
              return Sheets.m_110792_();
           } else {
-@@ -340,9 +_,70 @@
+@@ -340,12 +_,87 @@
        }
     }
  
 +   @Deprecated // FORGE: Use canRenderInLayer
     public static RenderType m_109287_(FluidState p_109288_) {
        RenderType rendertype = f_109276_.get(p_109288_.m_76152_());
        return rendertype != null ? rendertype : RenderType.m_110451_();
-+   }
-+
+    }
+ 
 +   // FORGE START
 +
 +   private static final java.util.function.Predicate<RenderType> SOLID_PREDICATE = type -> type == RenderType.m_110451_();
-+   private static final Map<net.minecraftforge.registries.IRegistryDelegate<Block>, java.util.function.Predicate<RenderType>> blockRenderChecks = createRenderCheckMap(f_109275_);
-+   private static final Map<net.minecraftforge.registries.IRegistryDelegate<Fluid>, java.util.function.Predicate<RenderType>> fluidRenderChecks = createRenderCheckMap(f_109276_);
-+
-+   private static <T extends net.minecraftforge.registries.ForgeRegistryEntry<T>> Map<net.minecraftforge.registries.IRegistryDelegate<T>, java.util.function.Predicate<RenderType>> createRenderCheckMap(
-+           Map<T, RenderType> vanillaMap
-+   ) {
-+      return Util.m_137469_(new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>(vanillaMap.size(), 0.5F), map -> {
-+         map.defaultReturnValue(SOLID_PREDICATE);
-+         for (Map.Entry<T, RenderType> entry : vanillaMap.entrySet()) {
-+            map.put(entry.getKey().delegate, createMatchingLayerPredicate(entry.getValue()));
-+         }
-+      });
-+   }
++   private static final RenderChecks<Block> BLOCK_RENDER_CHECKS = new RenderChecks<>(f_109275_);
```

Comment:
Please revert the name of these two maps to the previous camel case, otherwise this change will still break CTM because they are accessing this map via reflection

Reply:
are private forge-added fields considered stable api now? what a world we live in

Reply:
What is CTM depending on from these internals? Would it make sense in this PR to expand the API to handle CTM's use-case?

Reply:
They are retrieving the predicates from the map and, under certain conditions, adding them to their own map and overwriting the predicates stored in `ItemBlockRenderTypes` with their own.
https://github.com/Chisel-Team/ConnectedTexturesMod/blob/1.18/dev/src/main/java/team/chisel/ctm/client/util/CTMPackReloadListener.java#L64-L79
https://github.com/Chisel-Team/ConnectedTexturesMod/blob/1.18/dev/src/main/java/team/chisel/ctm/client/util/CTMPackReloadListener.java#L130-L136

Reply:
So it would seem that adding getters for the predicates should be enough?